src/referral/referral_view.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime, timedelta
+
+from rest_framework import status, viewsets
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from referral.related_models.referral_invite import ReferralInvite
+from referral.serializers.referral_invite_serializer import ReferralInviteSerializer
+from researchhub.settings import TESTING
+from user.related_models.user_model import User
+from utils.http import POST
+
+from .related_models.referral_invite import BOUNTY, JOIN_RH
+
+
+class ReferralViewSet(viewsets.ModelViewSet):
+    @action(detail=False, methods=[POST], permission_classes=[IsAuthenticated])
+    def invite(self, request):
+        inviter = request.user
+        data = request.data
+        recipient_email = data.get("email")
+        invite_type = data.get("type")
+        referral_first_name = data.get("first_name")
+        referral_last_name = data.get("last_name")
+        unified_document_id = data.get("unified_document_id", None)
```

Comment:
To ensure backend consistency, let's use the serializer to create the object for us. You can see an example in the create method of `bounty_view.py`.

Reply:
Makes sense. Updated.